liquidity_pool/src/lib.rs

```diff
@@ -181,7 +182,7 @@ impl LiquidityPoolTrait for LiquidityPool {
 
         let share_contract_id = create_contract(&e, &token_wasm_hash, &token_a, &token_b);
         token::Client::new(&e, &share_contract_id).initialize(
-            &get_contract_id(&e),
+            &e.current_contract_address(),
             &7u32,
             &Bytes::from_slice(&e, b"name"),
```

Comment:
nit: can we give these more meaningful values?

Reply:
done